src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -1300,6 +1302,7 @@ private bool ConsumeString()
                     ValueIsEscaped = false;
                     _tokenType = JsonTokenType.String;
                     _consumed += idx + 2;
+                    _hasPartialStringValue = false;
```

Comment:
This flags is introducing a new state for `Utf8JsonReader` that could break user-defined converters not expecting it. We should make absolutely sure instances setting this flag should never leak via public APIs. I would recommend adding a few debug asserts e.g. in the `Read()` method checking that the flag is always unset.